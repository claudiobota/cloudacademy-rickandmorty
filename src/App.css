body {
    background-color: #021b35;
}

.container {
    width: 100%;
    max-width: 1200px;
    margin-left: auto;
    margin-right: auto;
}
