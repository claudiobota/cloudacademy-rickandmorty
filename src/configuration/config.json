{
	"api": {
		"url": "https://rickandmortyapi.com/api",
		"paths": {
			"characters": "/character",
			"locations": "/location",
			"episodes": "/episode"
		},
		"page_size": 20
	}
}
