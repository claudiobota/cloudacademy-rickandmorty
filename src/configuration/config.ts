import configData from './config.json';
import configDevelopmentData from './config.development.json';
import configTestData from './config.test.json';
import configStagingData from './config.staging.json';
import configProductionData from './config.production.json';
import {IConfig} from '../types/interfaces';
import {Environment} from '../types/enums';

let config: IConfig = {
  ...configData
};

const nodeEnv: string = process.env.NODE_ENV || 'development';

switch (nodeEnv) {
  case Environment.DEVELOPMENT:
    config = {
      ...config,
      ...configDevelopmentData
    };
    break;

  case Environment.TEST:
    config = {
      ...config,
      ...configTestData
    };
    break;

  case Environment.STAGING:
    config = {
      ...config,
      ...configStagingData
    };
    break;

  case Environment.PRODUCTION:
    config = {
      ...config,
      ...configProductionData
    };
    break;

  default:
}

export default config;
