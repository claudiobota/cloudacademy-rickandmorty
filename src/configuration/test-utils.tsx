import React from 'react';
import {render as rtlRender} from '@testing-library/react';
import {Provider} from 'react-redux';
import {configureAppStore} from '../redux/store';

function render(
  ui: React.ReactElement,
  {
    initialState,
    store = configureAppStore(initialState),
    ...renderOptions
  }: any = {}
) {
  function Wrapper({ children }: { children: React.ReactElement }) {
    return <Provider store={store}>{children}</Provider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { render }
