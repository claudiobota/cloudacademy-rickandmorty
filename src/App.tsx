import React from 'react';
import './App.css';
import 'primeflex/primeflex.css';
import 'primereact/resources/themes/bootstrap4-light-blue/theme.css';
import 'primereact/resources/primereact.min.css';
import 'primeicons/primeicons.css';
import {BrowserRouter, Route, Switch} from 'react-router-dom';
import Home from './screens/Home/Home';
import Location from './screens/Location/Location';
import {Provider} from 'react-redux';
import store from './redux/store';
import {fetchAllEpisodes} from './redux/slices/episodeSlice';
import Error from './screens/Error/Error';

store.dispatch(fetchAllEpisodes());

function App() {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <Switch>
          <Route path="/location/:id" exact={true}>
            <Location />
          </Route>
          <Route path="/" exact={true}>
            <Home />
          </Route>
          <Route path="*" exact={true}>
            <Error />
          </Route>
        </Switch>
      </BrowserRouter>
    </Provider>
  );
}

export default App;
