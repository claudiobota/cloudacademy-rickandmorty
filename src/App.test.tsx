import React from 'react';
import App from './App';
import {render} from './configuration/test-utils';

describe('App', () => {
  test('app renders and matches snapshot', () => {
    const {asFragment} = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
