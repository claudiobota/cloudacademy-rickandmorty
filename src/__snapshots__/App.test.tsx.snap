// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App app renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="wrapper"
    >
      <div
        class="p-paginator p-component p-ml-auto p-mr-auto p-mb-5"
      >
        <button
          class="p-paginator-first p-paginator-element p-link p-disabled"
          disabled=""
          type="button"
        >
          <span
            class="p-paginator-icon pi pi-angle-double-left"
          />
        </button>
        <button
          class="p-paginator-prev p-paginator-element p-link p-disabled"
          disabled=""
          type="button"
        >
          <span
            class="p-paginator-icon pi pi-angle-left"
          />
        </button>
        <span
          class="p-paginator-pages"
        >
          <button
            class="p-paginator-page p-paginator-element p-link p-highlight"
            type="button"
          >
            1
          </button>
        </span>
        <button
          class="p-paginator-next p-paginator-element p-link p-disabled"
          disabled=""
          type="button"
        >
          <span
            class="p-paginator-icon pi pi-angle-right"
          />
        </button>
        <button
          class="p-paginator-last p-paginator-element p-link p-disabled"
          disabled=""
          type="button"
        >
          <span
            class="p-paginator-icon pi pi-angle-double-right"
          />
        </button>
      </div>
      <div
        class="p-grid p-align-stretch"
      />
      <div
        class="p-paginator p-component p-ml-auto p-mr-auto p-mt-5"
      >
        <button
          class="p-paginator-first p-paginator-element p-link p-disabled"
          disabled=""
          type="button"
        >
          <span
            class="p-paginator-icon pi pi-angle-double-left"
          />
        </button>
        <button
          class="p-paginator-prev p-paginator-element p-link p-disabled"
          disabled=""
          type="button"
        >
          <span
            class="p-paginator-icon pi pi-angle-left"
          />
        </button>
        <span
          class="p-paginator-pages"
        >
          <button
            class="p-paginator-page p-paginator-element p-link p-highlight"
            type="button"
          >
            1
          </button>
        </span>
        <button
          class="p-paginator-next p-paginator-element p-link p-disabled"
          disabled=""
          type="button"
        >
          <span
            class="p-paginator-icon pi pi-angle-right"
          />
        </button>
        <button
          class="p-paginator-last p-paginator-element p-link p-disabled"
          disabled=""
          type="button"
        >
          <span
            class="p-paginator-icon pi pi-angle-double-right"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
